torch/_inductor/triton_heuristics.py

```diff
@@ -329,14 +363,29 @@ def launcher({', '.join(def_args)}, grid, stream):
                 else:
                     grid_0, grid_1, grid_2 = grid
 
+                if hasattr(bin, "run"):
+                    runner = bin.run
+                else:
+                    runner = bin.c_wrapper
+
+                if hasattr(bin, "cluster_dims"):
+                    cluster_dims = bin.cluster_dims
+                else:
+                    cluster_dims = bin.clusterDims
+
+                if hasattr(bin, "function"):
+                    function = bin.function
+                else:
+                    function = bin.cu_function
+
                 if hasattr(bin, "num_ctas"):
-                    bin.c_wrapper(grid_0, grid_1, grid_2, bin.num_warps,
-                                bin.num_ctas, *bin.clusterDims, bin.shared,
-                                stream, bin.cu_function, None, None, None,
+                    runner(grid_0, grid_1, grid_2, bin.num_warps,
+                                bin.num_ctas, *cluster_dims, bin.shared,
+                                stream, function, None, None, None,
                                 {', '.join(call_args)})
                 else:
-                    bin.c_wrapper(grid_0, grid_1, grid_2, bin.num_warps, bin.shared,
-                                stream, bin.cu_function, None, None, None,
+                    runner(grid_0, grid_1, grid_2, bin.num_warps, bin.shared,
+                                stream, function, None, None, None,
```

Comment:
Let's do the version check at compile time and emit smaller/faster launcher code.  No need to call `hasattr` a bunch of times on every single kernel launch.